### PR TITLE
Admin user toggle

### PR DESCRIPTION
**Pull request by CoopTang**
- Extension 57/58: Site admin can enable/disable users.
- Disabled users cannot log in.
- Refactor tests to use string interpolation when dealing with links and IDs.

### CODE REVIEW

File: spec/features/admin/user_orders/show_spec.rb [hunk 1]
```diff
@@ -16,10 +16,9 @@
 
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(site_admin)
 
-      visit admin_user_show_path(user_1.id)
```
Comment (lrs8810) [inline spec/features/admin/user_orders/show_spec.rb:19]: Was this not working?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -37,4 +34,20 @@ def redirect_login
...
+  def user_login_redirect(user)
+    session[:user_id] = user.id
+    flash[:success] = "Welcome back, #{user.name}!"
+    redirect_login
+  end
+
+  def user_error_render
+    flash.now[:error] = 'The email and password you entered did not match our records. Please double-check and try again.'
+    render :login
+  end
```
Comment (freeheeling) [inline app/controllers/sessions_controller.rb:47]: I like how you elected to separate this out.

File: db/migrate/20191029222921_add_column_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddColumnToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :enabled?, :boolean, default: true
+  end
+end
```
Comment (freeheeling) [inline db/migrate/20191029222921_add_column_to_users.rb:5]: Was there an updated schema as well?

**Review by freeheeling — COMMENTED** [review]
All that refactoring – what a burden.

File: spec/features/admin/user_orders/show_spec.rb (hunk 1 above)
Comment (freeheeling) [reply spec/features/admin/user_orders/show_spec.rb:19]: We didn't  encounter any issues using this path helper in our testing.

**Pull request merged**